packages/editor/src/components/post-author/index.js

```diff
@@ -1,71 +1,137 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
 /**
  * WordPress dependencies
  */
+import { useState, useMemo } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { withInstanceId, compose } from '@wordpress/compose';
-import { Component } from '@wordpress/element';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { decodeEntities } from '@wordpress/html-entities';
+import { ComboboxControl } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
-export class PostAuthor extends Component {
-	constructor() {
-		super( ...arguments );
+let latestAvailableAuthors;
+function PostAuthor() {
+	const authors = useSelect( ( select ) => select( 'core' ).getAuthors() );
+	const postAuthor = useSelect( ( select ) =>
+		select( 'core' ).getAuthor(
+			select( 'core/editor' ).getEditedPostAttribute( 'author' )
+		)
+	);
+
+	const dispatch = useDispatch();
+
+	const authorsForField = useMemo( () => {
+		return authors.map( ( author ) => {
+			return {
+				key: author.id,
+				name: author.name,
+				id: author.id,
```

Comment:
You don't need the `id` property if the `key` has the same value.

Reply:
👍 will remove. we need the key.

Reply:
Removed these.